Add documentation for setting a request timeout and attributes with `WebClient` (#3582)
line/armeria: site/src/pages/docs/client-timeouts.mdx

Sometimes, the default timeouts used by the Armeria clients do not suit a particular scenario well.
In these cases, you might want to override the timeout settings.

> ## Using <type://ClientBuilder>

Reviewer: 👍

           .build(HelloService.Iface.class);
```

## Using <type://WebClientRequestPreparation>

> You can override a response timeout in a request level using <type://WebClient#prepare()>.

Reviewer 2: ```suggestion
You can override response timeout and max response length at request level using
<type://WebClient#prepare()>.
```

```java
import com.linecopr.armeria.client.WebClient;

WebClient client = WebClient.of("http://example.com"); 
client.prepare()
      .get("/hello")

>       .responseTimeout(Duration.ofSeconds(responseTimeout))

Reviewer 2: ```suggestion
      .responseTimeout(Duration.ofSeconds(30))
      .maxResponseLength(1024)
```

           .build(HelloService.Iface.class);
```

## Using <type://WebClientRequestPreparation>

> You can override response timeout and max response length at request level using <type://WebClient#prepare()>.

Reviewer 2: You might want to update this PR once your `RequestOptions` PR is merged. So let me leave it unmerged until you're done with the update. :wink:

Author: Agreed. Let me update this once `RequestOptions` is merged.

           .build(HelloService.Iface.class);
```

> ## Using <type://RequestOptions>

Reviewer 2: Ditto - How about splitting this into two sections, one for using preparation and the other for using options?
